src/Simulation/Microsoft.Quantum.TargetGeneration/GenerateTarget.fs

```diff
@@ -0,0 +1,398 @@
+                                    Range = (QsPositionInfo.Zero, QsPositionInfo.Zero)
+                                }
+                                |> QsTupleItem
+                            match args with
+                            | QsTupleItem one -> 
+                                [| controls ; QsTupleItem one |] |> ImmutableArray.ToImmutableArray |> QsTuple
+                            | QsTuple many ->
+                                if many.Length = 0 
+                                then controls
+                                else many.Insert(0, controls) |> QsTuple
+                         | _ -> args
+            let (argName, argsInit, argNames) =
+            //TODO: diagnostics.
+                let name = function | ValidName n -> n.Value | InvalidName -> ""
+                let rec buildVariableName = function
+                    | QsTupleItem  one -> one.VariableName |> name
+                    | QsTuple     many -> "(" + (many |> Seq.map buildVariableName |> String.concat ",") + ")"
+                let rec buildVariableNameList accum a =
+                    match a with
+                    | QsTupleItem  one -> ((one.VariableName |> name |> ``ident``) :: accum) |> List.rev
+                    | QsTuple     many -> many |> Seq.fold (fun s i -> buildVariableNameList s i) accum
+                match spArgs with
+                | QsTupleItem one -> (one.VariableName |> name, [], [one.VariableName |> name |> ``ident``])
+                | QsTuple many    ->
+                    if many.Length = 0 then
+                        ("__in__", [], [])
+                    elif many.Length = 1 then
+                        ((buildVariableName many.[0]), 
+                         [], 
+                         [``ident`` (buildVariableName many.[0])])
+                    else
+                        ("__in__",
+                         [ ``var`` (buildVariableName spArgs) (``:=`` <| inData) ], 
+                         buildVariableNameList [] spArgs)
+            let sim = ``((`` (``as`` targetClass ((``ident`` "this" ) <|.|> (``ident`` "Factory"))) ``))``
+            let methodName = implMethodName sp |> ``ident``
+            let call = sim <.> (methodName, argNames)
+            let ret =
+                match returnType.Resolution with
+                | QsTypeKind.UnitType ->
+                    [
+                        call |> statement
+                        ``return`` ( Some ((``ident`` "QVoid") <|.|> (``ident`` "Instance")) )
+                    ]
+                | _ ->
+                    [
+                        ``return`` ( Some call )
+                    ]
+            let body = ``() => {}`` [ argName ] (argsInit @ ret) :> ExpressionSyntax
+            Some (body |> propertyFromBody)
+        | Generated SelfInverse ->
+            let adjointedBodyName =
+                match sp.Kind with
+                | QsAdjoint           -> "Body"
```

Comment:
Same here, it would be good to get these from the C# gen.

Reply:
Done.